lib/ret_web/controllers/api/v1/media_controller.ex

```diff
@@ -10,29 +10,48 @@ defmodule RetWeb.Api.V1.MediaController do
     resolve_and_render(conn, url, 0)
   end
 
-  def create(conn, %{
-        "media" => %Plug.Upload{filename: filename, content_type: "application/octet-stream"} = upload,
-        "promotion_mode" => "with_token"
-      }) do
-    store_and_render_upload(conn, upload, MIME.from_path(filename), SecureRandom.hex())
+  def create(
+        conn,
+        %{"media" => %Plug.Upload{filename: filename, content_type: "application/octet-stream"} = upload} = params
+      ) do
+    convert_to_content_type = params |> Map.get("convert_to")
+    promotion_token = params |> promotion_token_for_params
+
+    store_and_render_upload(conn, upload, MIME.from_path(filename), convert_to_content_type, promotion_token)
   end
 
-  def create(conn, %{
-        "media" => %Plug.Upload{content_type: content_type} = upload,
-        "promotion_mode" => "with_token"
-      }) do
-    store_and_render_upload(conn, upload, content_type, SecureRandom.hex())
+  def create(conn, %{"media" => %Plug.Upload{content_type: content_type} = upload} = params) do
+    convert_to_content_type = params |> Map.get("convert_to")
+    promotion_token = params |> promotion_token_for_params
+
+    store_and_render_upload(conn, upload, content_type, convert_to_content_type, promotion_token)
   end
 
-  def create(conn, %{"media" => %Plug.Upload{filename: filename, content_type: "application/octet-stream"} = upload}) do
-    store_and_render_upload(conn, upload, MIME.from_path(filename))
+  defp promotion_token_for_params(%{"promotion_mode" => "with_token"}), do: SecureRandom.hex()
+  defp promotion_token_for_params(_params), do: nil
+
+  defp store_and_render_upload(conn, upload, content_type, convert_to_content_type, promotion_token)
+       when is_nil(convert_to_content_type) do
```

Comment:
Hmm, why use a guard here instead of just matching `nil = convert_to_content_type`?

Reply:
haha oops